compiler/rustc_codegen_gcc/src/consts.rs

```diff
@@ -322,13 +322,21 @@ pub fn const_alloc_to_gcc<'gcc, 'tcx>(cx: &CodegenCx<'gcc, 'tcx>, alloc: ConstAl
             )
             .expect("const_alloc_to_llvm: could not read relocation pointer")
             as u64;
+
+        let address_space = match cx.tcx.global_alloc(alloc_id) {
+            GlobalAlloc::Function(..) => cx.data_layout().instruction_address_space,
+            GlobalAlloc::Static(..) | GlobalAlloc::Memory(..) | GlobalAlloc::VTable(..) => {
+                AddressSpace::DATA
+            }
+        };
```

Comment:
this should probably be a method on `GlobalAlloc` (even if it has to take a `HasDataLayout` argument)

Reply:
This is actually copied from cg_llvm, so it is used in two places... (fixed in latest commit)